### DISCUSSION TITLE
dkim.KeyFormatError: Unparsable private key: Unexpected tag (got 30, expecting 02)

### DISCUSSION THREAD

**Discussion by alorson** [simple-login/app#1037] [Self-hosting question] [answered]
HI, 

i have installed docker and everything new and doubled checked everything (DNS, FILES, Conf). 
I can't register at the app because activation Mail not arriving. If i resend the activation mail, i will get Error 500 in Frontend. 

So i looked at the logs from sl-app. I found something related to dkim: 

```
dkim.KeyFormatError: Unparsable private key: Unexpected tag (got 30, expecting 02)
2022-05-31 11:01:22,398 - SL - DEBUG - "/code/server.py:444" - after_request() -  - 10.0.0.1 POST /auth/resend_activation ImmutableMultiDict([]) 500

dkim.asn1.ASN1FormatError: Unexpected tag (got 30, expecting 02)
```

```
root@simplelogin:~# ls -l
total 20
-rw-r--r-- 1 root root  916 May 31 10:58 dkim.key
-rw-r--r-- 1 root root  272 May 31 10:58 dkim.pub.key
-rw-r--r-- 1 root root  823 May 31 10:19 simplelogin.env

```

whats that about? I recreated the DKIM Keys and readded, rebooted and so on but nothing, same Error. 

Am i missing something?

**Comment by alorson** [comment, marked as answer]
Okay, 
i had openSSL 3.x on Ubuntu 22.x
i used locally OpenSSL 1.x and now DKIM Keys where accepted. Everything is working fine. 

THANK YOU.

**Reply by nguyenkims** [reply]
Glad that the issue is solved 👍

**Comment by akash07k** [comment]
So, can't we use openssl 3.X? I'm facing the same issue. am I required to use openssl 1.X?

**Comment by andryyy** [comment]
You can use the `-traditional` switch like this to create a `PKCS#1` key:

```
openssl genrsa -traditional -out privkey.pem 2048
```

Newer versions default to `PKCS#8`

**Comment by akash07k** [comment]
Thanks

**Comment by qimmiq** [comment]
awesome, i fixed my 502/500 error for registration and resend with that "-traditional" aswell
